api/sip/client.py

```diff
@@ -417,10 +420,20 @@ def make_secure_connection(self):
             fd, tmp_ssl_key_path = tempfile.mkstemp()
             os.write(fd, self.ssl_key.encode("utf-8"))
             os.close(fd)
-        connection = self.make_insecure_connection()
-        connection = ssl.wrap_socket(
-            connection, certfile=tmp_ssl_cert_path, keyfile=tmp_ssl_key_path
-        )
+        insecure_connection = self.make_insecure_connection()
+
+        context = self.ssl_contexts(ssl.PROTOCOL_TLS_CLIENT)
+        context.minimum_version = ssl.TLSVersion.TLSv1_2
+
+        # If the certificate path is provided, the certificate will be loaded.
+        # The private key is dependent on the certificate, so can't be loaded if
+        # there's no certificate present.
+        if tmp_ssl_cert_path:
+            context.load_cert_chain(
+                certfile=tmp_ssl_cert_path, keyfile=tmp_ssl_key_path
+            )
+
+        connection = context.wrap_socket(insecure_connection)
```

Comment:
I think this also has the happy side effect of wiping out one or more deprecation warnings. This is the new non-deprecated API.